dds_web/api/errors.py

```diff
@@ -21,10 +24,44 @@
 ####################################################################################################
 
 general_logger = logging.getLogger("general")
-action_logger = logging.getLogger("actions")
+action_logger = structlog.getLogger("actions")
 
 extra_info = {"result": "DENIED"}
 
+
+class LoggedHTTPException(exceptions.HTTPException):
+    """Base class to enable standard action logging on HTTP Exceptions"""
+
+    def __init__(self, message=None, **kwargs):
+
+        if auth.current_user():
+            current_user = auth.current_user().username
+        elif flask_login.current_user.is_authenticated:
+            current_user = flask_login.current_user.username
+        elif flask.request.remote_addr:
+            current_user = flask.request.remote_addr  # log IP instead of username
+        elif flask.request.access_route:
+            current_user = flask.request.access_route[0]  # log IP instead of username
+        else:
+            current_user = "anonymous"
```

Comment:
Since this is the same as in [here](https://github.com/ScilifelabDataCentre/dds_web/pull/771/files#diff-bacb9c85e5e141775ed3b7a626bf14730d2ec362398022c166bead23297eecc8R92-R101) maybe we should have a utils function for this?

Reply:
Sure. It also grew over time...